src/plugins/telemetry/server/telemetry_collection/get_local_license.ts

```diff
@@ -34,6 +34,52 @@ function fetchLicense(callCluster: LegacyAPICaller, local: boolean) {
   });
 }
 
+/**
+ * Get the cluster's license from the connected node.
+ *
+ * This is the equivalent of GET /_license?local=true&accept_enterprise=true .
+ *
+ * Like any X-Pack related API, X-Pack must installed for this to work.
+ */
+async function getLicenseFromLocalOrMasterNewClient(esClient: ElasticsearchClient) {
+  let response;
+  try {
+    // Fetching the license from the local node is cheaper than getting it from the master node and good enough
+    const { body } = await esClient.license.get<{ license: ESLicense }>({
+      local: true,
+      accept_enterprise: true,
+    });
+    cachedLicense = body.license;
+    response = body.license;
+  } catch (err) {
+    // if there is an error, try to get the license from the master node:
+    if (cachedLicense) {
+      try {
+        const { body } = await esClient.license.get<{ license: ESLicense }>({
+          local: false,
+          accept_enterprise: true,
+        });
+        cachedLicense = body.license;
+        response = body.license;
+      } catch (masterError) {
+        if (masterError.statusCode === 404) {
+          // the master node doesn't have a license and we assume there isn't a license
+          cachedLicense = undefined;
+          response = undefined;
+        } else {
+          throw err;
+        }
+      }
+    }
+    if (err.statusCode === 404) {
+      cachedLicense = undefined;
+    } else {
+      throw err;
+    }
+  }
+  return response;
+}
+
```

Comment:
Similar to the comment https://github.com/TinaHeiligers/kibana/pull/32#discussion_r485560395, I think it'll be easier to review if we modify the `fetchLicense` method to replace `callCluster('transport.request')` by `esClient.license.get`, and `getLocalLicense` to provide `esClient` instead of `callClient`.

Reply:
I'll go ahead and replace `callCluster` with `esClient` where we can and keep `bulkFetch` as suggested. Somehow my fork and rebasing are piling up commits, so I might need to draft another PR.